src/GitHubTools/Functions/Public/Update-AzurePipelineDependencies.ps1

```diff
@@ -0,0 +1,93 @@
+<#
+
+.SYNOPSIS
+Updates the refs of the azure-pipelines.yml file's repository resources to the latest
+
+.DESCRIPTION
+Updates the refs of the azure-pipelines.yml file's repository resources to the latest
+
+.PARAMETER GitHubOrganisation
+The GitHub organisation
+
+.PARAMETER RepositoryPrefix
+The prefix of repositories to create new branches with updated pipeline dependencies
+
+.EXAMPLE
+Set-GitHubSessionInformation -PatToken <not-a-real-pat-token>
+Update-AzurePipelineDependencies -GitHubOrganisation MyOrganisation -RepositoryPrefix foo-
+
+.NOTES
+powershell-yaml powershell module is required. Install-Module -name powershell-yaml
+The powershell-yaml module is no longer being maintained https://github.com/aaubry/YamlDotNet/discussions/689
+
+#>
+function Update-AzurePipelineDependencies {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "RepositoryPrefix", Justification = "False positive as rule does not know that Where-Object operates within the same scope")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseDeclaredVarsMoreThanAssignments", "UpdatePipelineYml", Justification="False positive as rule does not know that variable is updated in ForEach-Object")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseSingularNouns", "")]
+    param(
+        [Parameter(Mandatory = $false)]
+        [String]$GitHubOrganisation = "SkillsFundingAgency",
+        [Parameter(Mandatory = $false)]
+        [String]$RepositoryPrefix = "das-"
+    )
+
+    $Repos = Get-GitHubRepo -GitHubOrg $GitHubOrganisation | Where-Object { $_.name -match "^$RepositoryPrefix.+" -and $_.isArchived -eq $False }
+
+    $Repos | ForEach-Object {
+        $Repo = $_
+
+        Write-Host "Scanning $($Repo.name)"
+
+        $PipelineYml = Get-GitHubRepoFileContent -GitHubOrganisation $GitHubOrganisation -RepositoryName $Repo.name -FilePath "azure-pipelines.yml"
+
+        if (!$PipelineYml) { return }
+
+        $Pipeline = ConvertFrom-Yaml -Yaml $PipelineYml.Content
+
+        $UpdatePipelineYml = $false
+        $Pipeline.resources.repositories | Where-Object { $_.type -eq "github" } | ForEach-Object {
+            if ($_.ref -like "refs/tags/*") {
+                $TagName = $_.ref.replace("refs/tags/", "")
+                $DependencyOrg = $_.name.Split("/")[0]
+                $DependencyRepo = $_.name.Split("/")[1]
+                $Releases = Get-GitHubRepoRelease -GitHubOrganisation $DependencyOrg -RepositoryName $DependencyRepo
+                $OutOfDate = $Releases[0].tag_name -ne $TagName
+
+                if ($OutOfDate) {
+                    Write-Warning "$($_.name) dependency: $TagName should be $($Releases[0].tag_name): updating"
+                    $PipelineYml.Content = $PipelineYml.Content.Replace($TagName, $Releases[0].tag_name)
```

Comment:
This will replace any string that matches `$TagName`, eg `1.0.1` with the new value, this could be the tag of other resources, a comment or some other element of the pipeline.  I think we need to use a [capturing group](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comparison_operators?view=powershell-7.2#regular-expressions-substitutions) to do that, happy to help you work out how to do that :)

Reply:
Have explicitly filtered to only our two repositories with Where-Object